src/main/java/org/jitsi/jibri/capture/ffmpeg/FfmpegCapturer.kt

```diff
@@ -0,0 +1,42 @@
+package org.jitsi.jibri.capture.ffmpeg
+
+import org.jitsi.jibri.sink.Sink
+import org.jitsi.jibri.capture.Capturer
+import org.jitsi.jibri.capture.UnsupportedOsException
+import org.jitsi.jibri.capture.ffmpeg.executor.FfmpegExecutor
+import org.jitsi.jibri.capture.ffmpeg.executor.FfmpegExecutorParams
+import org.jitsi.jibri.capture.ffmpeg.executor.impl.LinuxFfmpegExecutor
+import org.jitsi.jibri.capture.ffmpeg.executor.impl.MacFfmpegExecutor
+import org.jitsi.jibri.util.extensions.debug
+import java.util.logging.Logger
+
+/**
+ * [FfmpegCapturer] is responsible for launching ffmpeg, capturing from the
+ * configured audio and video devices, and writing to the given [Sink]
+ */
+class FfmpegCapturer : Capturer {
+    private val logger = Logger.getLogger(this::class.qualifiedName)
+    val ffmpegExecutor: FfmpegExecutor
+
+    init
+    {
+        logger.debug("OS: ${System.getProperty("os.name")}")
+        ffmpegExecutor = when (System.getProperty("os.name")) {
+            "Mac OS X" -> MacFfmpegExecutor()
+            "Linux" -> LinuxFfmpegExecutor()
+            else -> throw UnsupportedOsException()
```

Comment:
I'm wondering if it's best to hide this completely in another factory so FfmpegCapturer doesn't have to know about OS. Should this file be called FfmpegCapturer or something more generic to hide the Ffmpeg detail?

Reply:
well `FfmpegCapturer` basically is that factory, this is where this logic is being hidden (it doesn't do much else).  And Ffmpeg in the name is intentional, this isn't the generic part (that's just `Capturer`) this is the specific type that denotes it as an _ffmpeg_ Capturer